keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java

```diff
@@ -538,26 +469,27 @@ public VoidResponse purgeDeletedKey(String name) {
      * @return A {@link Response} whose {@link Response#value() value} contains the {@link Key restored key}.
      */
     public Response<Key> restoreKey(byte[] backup) {
-        KeyRestoreRequestParameters parameters = new KeyRestoreRequestParameters().keyBackup(backup);
-        return service.restoreKey(endpoint, API_VERSION, parameters, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block();
+        return client.restoreKey(backup).block();
     }
 
     /**
      * List keys in the key vault. Retrieves a list of the keys in the Key Vault as JSON Web Key structures that contain the public
      * part of a stored key. The List operation is applicable to all key types and the individual key response in the list is represented by {@link KeyBase} as only the base key identifier,
      * attributes and tags are provided in the response. The key material and individual key versions are not listed in the response. This operation requires the {@code keys/list} permission.
      *
-     * <p>It is possible to get full keys with values from this information. Loop over the {@link KeyBase key} and
-     * call {@link KeyClient#getKey(KeyBase baseKey)} . This will return the {@link Key key} with value included of its latest version.</p>
+     * <p>It is possible to get full keys with key material from this information. Loop over the {@link KeyBase key} and
+     * call {@link KeyClient#getKey(KeyBase baseKey)} . This will return the {@link Key key} with key material included of its latest version.</p>
      * <pre>
-     * keyClient.listKeys().stream().map(keyClient::getKey).forEach(keyResponse -&gt;
-     *   System.out.printf("Received key with name %s and id %s", keyResponse.value().name(), keyResponse.value().id()));
+     * for (KeyBase key : keyClient.listKeys()) {
+     *   Key keyWithMaterial = keyClient.getKey(key).value();
+     *   System.out.printf("Received key with name %s and type %s", keyWithMaterial.name(), keyWithMaterial.keyMaterial().kty());
+     * }
      * </pre>
      *
      * @return A {@link List} containing {@link KeyBase key} of all the keys in the vault.
      */
-    public List<KeyBase> listKeys() {
-        return service.getKeys(endpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(r -> extractAndFetchKeys(r, Context.NONE)).collectList().block();
+    public Iterable<KeyBase> listKeys() {
+        return client.listKeys().collectList().block();
```

Comment:
Use Flux.toIterable: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#toIterable--

Reply:
updated in KeyClient and SecretClient list ops.